routingpy/routers/opentripplanner.py

```diff
@@ -0,0 +1,244 @@
+            timeout,
+            retry_timeout,
+            retry_over_query_limit,
+            skip_api_error,
+            **client_kwargs,
+        )
+
+    def directions(
+        self,
+        locations: List[List[float]],
+        profile: str,
+        alternatives: Optional[Union[bool, int]] = None,
+        dry_run: Optional[bool] = None,
+    ):
+        """
+        Get directions between an origin point and a destination point.
+
+        :param locations: The coordinates tuple the route should be calculated from in order of
+            visit.
+        :type locations: list of list
+
+        :param profile: Specifies the mode of transport to use when calculating directions. Possible
+            values are "CAR", "BICYCLE", "TRANSIT", "WALK". For more profiles, see OpenTripPlanner's
+            GraphiQL documentation.
+        :type profile: str
+
+        :param alternatives: Search for alternative routes. A result cannot be guaranteed. Accepts
+            an integer or False. Default False.
+        :type alternatives: bool
+
+        :param dry_run: Print URL and parameters without sending the request.
+        :type dry_run: bool
+
+        :returns: One or multiple route(s) from provided coordinates and restrictions.
+        :rtype: :class:`routingpy.direction.Direction` or :class:`routingpy.direction.Directions`
+        """
+        num_itineraries = 1 if not alternatives else 5
+        query = f"""
+            {{
+                plan(
+                    from: {{lat: {locations[0][1]}, lon: {locations[0][0]}}}
+                    to: {{lat: {locations[1][1]}, lon: {locations[1][0]}}}
+                    transportModes: [{{mode: {profile}}}]
+                    numItineraries: {num_itineraries}
+                ) {{
+                    itineraries {{
+                        duration
+                        legs {{
+                            duration
+                            distance
+                            legGeometry {{
+                                points
+                            }}
+                        }}
+                    }}
+                }}
+            }}
+        """
+        params = {"query": query}
+        response = self.client._request(
+            "/otp/routers/default/index/graphql", post_params=params, dry_run=dry_run
+        )
+        return self._parse_directions_response(response, alternatives)
+
+    def _parse_directions_response(self, response, alternatives):
+        routes = []
+        for itinerary in response["data"]["plan"]["itineraries"]:
+            geometry, distance = self._parse_legs(itinerary["legs"])
+            routes.append(
+                Direction(
+                    geometry=geometry,
+                    duration=int(itinerary["duration"]),
+                    distance=distance,
+                    raw=itinerary,
+                )
+            )
+
+        if alternatives:
+            return Directions(routes, raw=response)
```

Comment:
it was a stupid decision to return multiple types from the same function. def not your fault, just expressing my general dissatisfaction with my past self;)

Reply:
Yes I think we could simplify the API in a future version.